File: Data Structures & Algorithms/buy-and-sell-crypto/submission-6.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) is 1:
            return 0;

        left = 0;
        right = left + 1;
        maxProfit = 0;

        while (right < len(prices)):
            if (prices[right] - prices[left] < 0):
                left = right;
                right += 1;
            else:
                if (prices[right] - prices[left] > maxProfit):
                    maxProfit = prices[right] - prices[left];
                right += 1;

        return maxProfit

        

File: Data Structures & Algorithms/is-palindrome/submission-6.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();

        for (char c : s.toLowerCase().toCharArray()) {
            if (Character.isLetterOrDigit(c)) {
                sb.append(c);
            }
        }

        String newS = sb.toString();
        System.out.println(newS);

        int left = 0;
        int right = newS.length() - 1;

        while (right > left ) {
            if (newS.charAt(left) != newS.charAt(right)) {
                return false;
            }
            left ++;
            right --;
        }
        return true;

    }
}
